### PR TITLE
Add `domain_data` to Organization API and deprecate `domains`

### PR DESCRIPTION
**Pull request by oliverzheng**
## Documentation

Does this require changes to the WorkOS Docs? E.g. the [API Reference](https://workos.com/docs/reference) or code snippets need updates.

```
[ ] Yes
```

If yes, link a related docs PR and add a docs maintainer as a reviewer. Their approval is required.

### CODE REVIEW

File: lib/Organizations.php [hunk 1]
```diff
@@ -93,19 +95,20 @@ public function createOrganization($name, $domains, $allowProfilesOutsideOrganiz
...
+     * @param array $domains [Deprecated] The domains of the Organization. Use domain_data instead.
+     * @param array $domain_data The domains of the Organization.
      * @param string $name The name of the Organization.
      * @param null|boolean $allowProfilesOutsideOrganization Whether Connections within the Organization allow profiles
      *      that are outside of the Organization's configured User Email Domains.
      *
      * @throws Exception\WorkOSException
      */
-    public function updateOrganization($organization, $domains, $name, $allowProfilesOutsideOrganization = null)
+    public function updateOrganization($organization, $domains = null, $name = null, $allowProfilesOutsideOrganization = null, $domain_data = null)
```
Comment (oliverzheng) [inline lib/Organizations.php:106]: `domains` is not be required to update an organization.

File: lib/Organizations.php [hunk 2]
```diff
@@ -93,19 +95,20 @@ public function createOrganization($name, $domains, $allowProfilesOutsideOrganiz
...
     {
         $organizationsPath = "organizations/{$organization}";
         $params = [
-          "organization" => $organization,
```
Comment (oliverzheng) [inline lib/Organizations.php:107]: The organization ID is embedded in the path and is not a param.

**Comment by mthadley** [comment]
@workos/php I'll be taking over this PR, but if one of y'all can give this a review, it'd be much appreciated!

File: lib/Organizations.php [hunk 3]
```diff
@@ -65,7 +65,8 @@ public function listOrganizations(
      * Create Organization.
      *
      * @param string $name The name of the Organization.
-     * @param array $domains The domains of the Organization.
+     * @param null|array $domains [Deprecated] The domains of the Organization. Use domain_data instead.
```
Comment (PaulAsjes) [inline lib/Organizations.php:68]: We should add something along the lines of "Will be removed in a future major version" to further encourage people not to use this.

File: lib/Organizations.php [hunk 4]
```diff
@@ -93,19 +95,20 @@ public function createOrganization($name, $domains, $allowProfilesOutsideOrganiz
      * Update Organization.
      *
      * @param string $organization An Organization identifier.
-     * @param array $domains The domains of the Organization.
-     * @param string $name The name of the Organization.
+     * @param null|array $domains [Deprecated] The domains of the Organization. Use domain_data instead.
```
Comment (PaulAsjes) [inline lib/Organizations.php:98]: Same comment as above.

File: lib/Organizations.php (hunk 2 above)
Comment (PaulAsjes) [reply lib/Organizations.php:107]: Is this safe to remove? Will we break an existing user's integration? I'd prefer if this were a minor version release.

**Review by PaulAsjes — CHANGES_REQUESTED** [review]

File: lib/Organizations.php (hunk 3 above)
Comment (mthadley) [reply lib/Organizations.php:68]: That's generally what "Deprecated" means to me, but no harm in adding some more context if you think it'd be useful.

File: lib/Organizations.php (hunk 2 above)
Comment (mthadley) [reply lib/Organizations.php:107]: Yep, this is safe. This parameter was never needed (for the reason Oliver mentioned) and won't introduce any breakage.

File: lib/Organizations.php (hunk 3 above)
Comment (PaulAsjes) [reply lib/Organizations.php:68]: I find it better to be extra explicit in these cases, not a big deal though.

**Pull request merged**